### PR TITLE
rewrite and simplify the resource monitor package

### PR DESCRIPTION
**Pull request by fromanirh**
Reduce the API surface as much as possible in order to simplify the package and trim down the size.
We get rid of the intermediate representation, with all the extra complexity this encompesses.
    
Along the way, we add the option to refresh the allocatable resources before each polling cycle,
instead of just once at the initialization time.

### CODE REVIEW

**Review by swatisehgal — COMMENTED** [review]
Thanks for working on this major refactoring of the code. Combining the scanning and aggregation logic seems like a good idea and overall the changes look good to me. The only minor comment I have is that since we have moved all the code to `resourcemonitor.go` and it seems like a lot squeezed into a single file. Maybe we keep excludeList logic in a separate file and datatype definitions in the type.go like it was previously? Let me know if that sounds reasonable.

File: pkg/nrtupdater/nrtupdater.go [hunk 1]
```diff
@@ -139,3 +139,11 @@ func (te *NRTUpdater) Run(infoChannel <-chan MonitorInfo) chan<- struct{} {
 	}()
 	return done
 }
+
+func dumpObject(obj interface{}) string {
```
[the remark this answers is not in the document]
Comment (fromanirh) [reply pkg/nrtupdater/nrtupdater.go:143]: thanks, fixed!

**Comment by fromanirh** [comment]
> The only minor comment I have is that since we have moved all the code to `resourcemonitor.go` and it seems like a lot squeezed into a single file. Maybe we keep excludeList logic in a separate file and datatype definitions in the type.go like it was previously? Let me know if that sounds reasonable.

I'm not sure. We extract indeed some components, but these components will be tiny and used all over the place in the bigger file. I'm not sure this modularization actually improves redability.

**Comment by swatisehgal** [comment]
That is indeed a valid point, having to go back and forth to other files will impact the readability. Let's leave it like this for now and see what the maintainers think. I am happy enough with the PR!

/lgtm

**Comment by fromanirh** [comment]
> That is indeed a valid point, having to go back and forth to other files will impact the readability. Let's leave it like this for now and see what the maintainers think. I am happy enough with the PR!
> 
> /lgtm

thanks! :)

**Pull request merged**